segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/handler/AppendProcessorTest.java

```diff
@@ -161,6 +168,33 @@ public void testAppend() {
         assertTrue(processor.isSetupAppendCompleted(setupAppendCommand.getSegment(), setupAppendCommand.getWriterId()));
     }
 
+    @Test
+    public void testCreateTransientSegment() throws Exception {
+
+        String parentSegmentName = "scope/stream/parentSegment";
+        UUID writerId = UUID.randomUUID();
+        @Cleanup
+        ServiceBuilder serviceBuilder = newInlineExecutionInMemoryBuilder(getBuilderConfig());
+        serviceBuilder.initialize();
+        StreamSegmentStore store = serviceBuilder.createStreamSegmentService();
+
+        ServerConnection connection = mock(ServerConnection.class);
+        ConnectionTracker tracker = mock(ConnectionTracker.class);
+        val mockedRecorder = Mockito.mock(SegmentStatsRecorder.class);
+        AppendProcessor processor = AppendProcessor.defaultBuilder()
```

Comment:
Add `@Cleanup` to this instance (which is essentially a `try-with-resources+finally`) and to every other place in this file (even the ones you haven't touched) where `AppendProcessor` is instantiated.